Adding gRPC overview beta doc and updating few pages to direct attention to relevant data access content (#21722)
MystenLabs/sui: docs/content/concepts/grpc-overview.mdx

---
title: gRPC Overview (Beta)
description: Overview of the gRPC API to access Sui network data.
beta: devnet, testnet, mainnet
---

The [Sui Full Node gRPC API](../references/fullnode-protocol.mdx) provides a fast, type-safe, and efficient interface for interacting with the Sui blockchain. Designed for power users, indexers, explorers, and decentralized apps, this API enables access to Sui data with high performance and low latency.

:::info

{@include: ../snippets/data-serving-msg.mdx}

:::

## What is gRPC?

> gRPC offers a high-performance, efficient communication protocol that uses [Protocol Buffers](https://protobuf.dev/overview/) for fast, compact data serialization. Its strongly typed interfaces reduce runtime errors and simplify client/server development across multiple languages. With built-in support for code generation, you can quickly scaffold clients in Typescript, Go, Rust, and more. This makes it ideal for scalable backend systems like indexers, blockchain explorers, and data-intensive decentralized apps.

Reviewer: I would avoid spelling it out entirely. Officially it stands for:

gRPC (acronym for gRPC Remote Procedure Calls)

> gRPC offers a high-performance, efficient communication protocol that uses [Protocol Buffers](https://protobuf.dev/overview/) for fast, compact data serialization. Its strongly typed interfaces reduce runtime errors and simplify client/server development across multiple languages. With built-in support for code generation, you can quickly scaffold clients in Typescript, Go, Rust, and more. This makes it ideal for scalable backend systems like indexers, blockchain explorers, and data-intensive decentralized apps.

Author: Keeping it as is per Brandon's comment

```shell
grpc.health.v1.Health
grpc.reflection.v1.ServerReflection
sui.rpc.v2alpha.LiveDataService
sui.rpc.v2alpha.SubscriptionService
sui.rpc.v2beta.LedgerService
sui.rpc.v2beta.TransactionExecutionService
```

**List available APIs in the LedgerService**

```shell
grpcurl <full node URL:port> list sui.rpc.v2beta.LedgerService
```

which should return something like:

```shell
sui.rpc.v2beta.LedgerService.BatchGetObjects
sui.rpc.v2beta.LedgerService.BatchGetTransactions
sui.rpc.v2beta.LedgerService.GetCheckpoint
sui.rpc.v2beta.LedgerService.GetEpoch
sui.rpc.v2beta.LedgerService.GetObject
sui.rpc.v2beta.LedgerService.GetServiceInfo
sui.rpc.v2beta.LedgerService.GetTransaction
```

**Get the `events` and `effects` details of a particular transaction**

```shell
grpcurl -d '{ "digest": "3ByWphQ5sAVojiTrTrGXGM5FmCVzpzYmhsjbhYESJtxp" }' <full node URL:port> sui.rpc.v2beta.LedgerService/GetTransaction
```

**Get the transactions in a particular checkpoint**

```shell
grpcurl -d '{ "sequence_number": "180529334", "read_mask": { "paths": ["transactions"]} }' <full node URL:port> sui.rpc.v2beta.LedgerService/GetCheckpoint
```

### Access using typescript

> This is an example to build a typescript client for Sui gRPC API. Feel free to use another set of tools or modules that you’re comfortable with.

Reviewer 2: ```suggestion
This is an example to build a TypeScript client for Sui gRPC API. If you want to use a different set of tools or modules that you’re comfortable with, you can adjust the instructions accordingly.
```

**Install dependencies**

> ```shell

Reviewer 2: Put each of these in their own codeblock. The site has a copy code button, which fails if the user tries to copy and run as is when there are multiple commands. And use `sh` instead. And previously mentioned $ sign.
```
'''sh
$ npm init -y
'''
'''sh
$ npm install...
'''
```

npm init -y
npm install @grpc/grpc-js @grpc/proto-loader
npm i -D tsx
```

**Project structure**

```shell
.
├── protos/
│   └── sui/
│       └── node/
│           └── v2beta/
│               ├── ledger_service.proto
│               └── *.proto
├── client.ts
├── package.json
```

> Download all the sui/rpc/v2beta proto files from [Github v2beta](https://github.com/MystenLabs/sui/tree/main/crates/sui-rpc-api/proto) in the same folder.

Reviewer 2: ```suggestion
Download all the `sui/rpc/v2beta proto` files from [Github v2beta](https://github.com/MystenLabs/sui/tree/main/crates/sui-rpc-api/proto) in the same folder.
```

**Sample client.ts to get `events` and `effects` details of a particular transaction**

> ```typescript

Reviewer 2: ```suggestion
```ts
```
